Backend/routes/routes.py

```diff
@@ -405,54 +412,88 @@ async def delete_supplement_details(id: str):
 
 
-@program_router.post("/programs", response_model=Program)
-async def create_program(program: Program):
-    program_dict = program.dict()
-    program_dict["donor"] = {"_id": ObjectId(
-        program.donor.id), "name": program.donor.name}
-    program_dict["supplement"] = {"_id": ObjectId(
-        program.supplement.id), "name": program.supplement.name}
-    result = ProgramsCollection.insert_one(program_dict)
-    program.id = str(result.inserted_id)
-    return program
-
-
-@program_router.get("/programs", response_model=List[Program])
-async def read_programs():
-    programs = []
-    for program_dict in ProgramsCollection.find():
-        program = Program.from_dict(program_dict)
-        programs.append(program)
-    return programs
-
-
-@program_router.get("/programs/{program_id}", response_model=Program)
-async def read_program(program_id: str):
-    program_dict = ProgramsCollection.find_one({"_id": ObjectId(program_id)})
-    if program_dict is None:
-        raise HTTPException(status_code=404, detail="Program not found")
-    program = Program.from_dict(program_dict)
-    return program
-
-
-@program_router.put("/programs/{program_id}", response_model=Program)
-async def update_program(program_id: str, program: Program):
-    program_dict = program.dict()
-    program_dict["donor"] = {"_id": ObjectId(
-        program.donor.id), "name": program.donor.name}
-    program_dict["supplement"] = {"_id": ObjectId(
-        program.supplement.id), "name": program.supplement.name}
-    result = ProgramsCollection.replace_one(
-        {"_id": ObjectId(program_id)}, program_dict)
-    if result.matched_count == 0:
-        raise HTTPException(status_code=404, detail="Program not found")
-    program.id = program_id
-    return program
-
-
-@program_router.delete("/programs/{program_id}")
-async def delete_program(program_id: str):
-    result = ProgramsCollection.delete_one({"_id": ObjectId(program_id)})
-    if result.deleted_count == 0:
-        raise HTTPException(status_code=404, detail="Program not found")
-    return {"message": "Program deleted successfully"}
+@program_router.post("/api/add_program")
+async def add_program(program: Program):
+    donor_details = DonorsCollection.find_one({"_id": ObjectId(program.donor_id)})
+    suplement_details = SupplementDetailsCollection.find_one({"_id": ObjectId(program.supplements_details_id)})
+    if not donor_details and suplement_details:
+        raise HTTPException(status_code=404, detail="input are missing")
+    _id = ProgramsCollection.insert_one(dict(program))
+    added_programs = program_list_serializer(ProgramsCollection.find(
+        {"_id": _id.inserted_id}))
+    return {"status": "ok", "data": added_programs}
+
+
+@program_router.get("/api/get_programs")
+async def get_programs():
+    programs = program_list_serializer(
+        ProgramsCollection.find())
+    return {"status": "ok", "data": programs}
+
+
+@program_router.get("/api/get_program/{program_id}")
```

Comment:
in url it is "program_id" but in parameter it is given as "id" will it work?

Reply:
it will work the value will take from the parameter but let me change the name